eclair-core/src/main/scala/fr/acinq/eclair/crypto/keymanager/LocalChannelKeyManager.scala

```diff
@@ -38,8 +38,25 @@ object LocalChannelKeyManager {
 }
 
 /**
- * This class manages channel secrets and private keys.
- * It exports points and public keys, and provides signing methods
+ * An implementation of [[ChannelKeyManager]] that supports deterministic derivation of keys, based on the initial
+ * funding pubkey.
+ *
+ * Specifically, there are two paths both of length 8 (256 bits):
+ *   - `fundingKeyPath`: chosen at random using `newFundingKeyPath()`
+ *   - `channelKeyPath`: sha(fundingPubkey(0)) using `ChannelKeyManager.keyPath()`
+ *
+ * The resulting paths looks like so on mainnet:
+ * {{{
+ *  funding:
+ *   - initial funding tx (*):
+ *     47' / 1' / <fundingKeyPath> / <1' or 0'> / 0'
+ *   - splice funding txs (index > 0):
+ *     47' / 1' / <fundingKeyPath> / <1' or 0'> / 6' / <index>
```

Comment:
I think we shouldn't add the unnecessary `/ 6' /` and treat splice txs exactly the same as the initial funding tx:

```suggestion
 *     47' / 1' / <fundingKeyPath> / <1' or 0'> / <index'>
```

This way in `override def fundingPublicKey` we don't need to treat the initial funding pubkey differently and can do: `internalKeyPath(channelKeyPath, hardened(fundingTxIndex))`

This guarantees that even if the private key for an old `fundingTxIndex` is leaked, it doesn't leak the private key of the current `fundingTxIndex`, which is nice!

Reply:
Done in a38095a5f.